### ISSUE TITLE
support for PKCS8 encrypted 4096 bit key

### ISSUE THREAD

**Issue by RonEld** [ARMmbed/mbedtls#1019]
As raised in the [mbed TLS support forum](https://tls.mbed.org/discussions/bug-report-issues/buffer-too-small-to-parse-encrypted-private-key) by Florian, 4096 bit encrypted key is not supported.
This is because the key buffer size for this format is 2376 bytes, and the internal buffer used for the decryption is 2048 bytes long.
Error returned by `pk_parse_key_pkcs8_encrypted_der` is `MBEDTLS_ERR_PK_BAD_INPUT_DATA` because:
~~~
    if( len > sizeof( buf ) )
        return( MBEDTLS_ERR_PK_BAD_INPUT_DATA );
~~~

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-1566

**Comment by RonEld** [comment]
similar to #878

**Comment by gilles-peskine-arm** [comment]
Resolved by #1062 which is now merged.
